contracts/pcv/maple/MaplePCVDeposit.sol

```diff
@@ -0,0 +1,354 @@
+        int256 balanceAfterPnl = lastRecordedBalanceAfterPnl.toInt256();
+
+        /// withdraw from the pool
+        /// this call will withdraw amount of principal requested, and then send
+        /// over any accrued interest.
+        /// expected behavior is that this contract
+        /// receives either amount of USDC, or amount of USDC + interest accrued
+        /// if lending losses were taken, receive less than amount
+        IMaplePool(maplePool).withdraw(amount);
+
+        /// withdraw min between balance and amount as losses could be sustained in venue
+        /// causing less than amt to be withdrawn
+        uint256 amountToTransfer = Math.min(
+            IERC20(USDC).balanceOf(address(this)),
+            amount
+        );
+        IERC20(USDC).safeTransfer(to, amountToTransfer);
+
+        uint256 balanceAfterTransfer = lastRecordedBalanceAfterPnl -
+            amountToTransfer;
+        lastRecordedBalance = balanceAfterTransfer;
+
+        if (pcvOracle != address(0)) {
+            /// if any amount of PCV is withdrawn and no gains, delta is negative
+            IPCVOracle(pcvOracle).updateIlliquidBalance(
+                balanceAfterTransfer.toInt256() - balanceBeforePnl
+            );
+        }
+
+        emit Withdrawal(msg.sender, to, amountToTransfer);
+    }
+
+    /// ---------- Sad Path APIs ----------
+
+    /// Assume that using these functions will likely
+    /// break all happy path functions
+    /// Only use these function in an emergency situation
+
+    /// -----------------------------------
+
+    /// @notice get rewards and unstake from rewards contract
+    /// breaks functionality of happy path withdraw functions
+    function exitRewards() external onlyPCVController {
+        IMapleRewards(mapleRewards).exit();
+    }
+
+    /// @notice unstake from rewards contract without getting rewards
+    /// breaks functionality of happy path withdraw functions
+    function withdrawFromRewardsContract() external onlyPCVController {
+        uint256 rewardsBalance = IMaplePool(maplePool).balanceOf(address(this));
+        IMapleRewards(mapleRewards).withdraw(rewardsBalance);
+    }
+
+    /// @notice unstake from Pool FDT contract without getting rewards
+    /// or unstaking from the reward contract.
+    /// @param to destination after funds are withdrawn from venue
+    /// @param amount of PCV to withdraw from the venue
+    function withdrawFromPool(address to, uint256 amount)
+        external
+        onlyPCVController
+    {
+        IMaplePool(maplePool).withdraw(amount);
+        /// withdraw min between balance and amount as losses could be sustained in venue
+        /// causing less than amt to be withdrawn
+        uint256 amountToTransfer = Math.min(
+            IERC20(USDC).balanceOf(address(this)),
+            amount
+        );
+        IERC20(USDC).safeTransfer(to, amountToTransfer);
+
+        emit Withdrawal(msg.sender, to, amountToTransfer);
+    }
+
```

Comment:
this should have a title above it indicating it is not part of the sad path and is just a helper function.

Reply:
like this https://github.com/volt-protocol/volt-protocol-core/pull/138/files#diff-e06d47fadf72a039b820fbb3fbae0c2071ac2ccee06d6d25f325262fc494020dR283

Reply:
I moved it above the "sad path functions"